torch/_functorch/partitioners.py

```diff
@@ -1494,21 +1495,50 @@ def dp_knapsack(
     # The maximum runtime that can be achieved within the max_memory constraint
     max_runtime = dp[n][quantized_max_memory].item()
 
+    print("SAVED VALUES:")
+    print(saved_items)
+    print("RECOMP VALUES:")
+    print(recomputable_items)
     return max_runtime, saved_items, recomputable_items
 
 
 def _optimize_runtime_with_given_memory(
+    joint_graph: fx.Graph,
     memory: List[float],
     runtimes: List[float],
     max_memory: float,
+    node_info: NodeInfo,
+    all_recomputable_banned_nodes: List[fx.Node],
 ) -> Tuple[float, List[int], List[int]]:
     SOLVER = config.activation_memory_budget_solver
     if SOLVER == "greedy":
+        memory = list(memory.values())
         return greedy_knapsack(memory, runtimes, max_memory)
     elif SOLVER == "ilp":
+        memory = list(memory.values())
         return ilp_knapsack(memory, runtimes, max_memory)
     elif SOLVER == "dp":
+        memory = list(memory.values())
         return dp_knapsack(memory, runtimes, max_memory)
+    elif SOLVER == "backward_pass_aware":
+        try:
+            print("----- solving ILP based approach ----")
+            saved_node_idx, recomp_node_idx = get_saved_recomp_nodes_from_BWPA(
+                memory,
+                joint_graph,
+                max_memory,
+                node_info,
+                all_recomputable_banned_nodes,
+            )
+            return [0.0, saved_node_idx, recomp_node_idx]
+        except Exception as e:
+            print(f"ILP based approach failed with error: {e}")
+            print("----- solving DP based approach ----")
+            memory = list(memory.values())
+            return dp_knapsack(memory, runtimes, max_memory)
```

Comment:
Why does this fail?  We should fix tese failures rather than silently ignoring them.